nixos/modules/services/networking/prosody.nix

```diff
@@ -255,37 +417,47 @@ in
 
   config = mkIf cfg.enable {
 
-    environment.systemPackages = [ pkgs.prosody ];
+    environment.systemPackages = [ cfg.package ];
 
     environment.etc."prosody/prosody.cfg.lua".text = ''
```

Comment:
I suggest using the `--config` option instead of polluting `/etc`.

Reply:
We would have to create a specially wrapped prosodyctl then, so :-1:

Reply:
True. Didn't think about that because I don't use `prosodyctl` for managing users.